packages/controller/src/WsServerConnector.ts

```diff
@@ -114,7 +114,7 @@ export default class WsServerConnector extends lib.WebSocketBaseConnector {
 	_timedOut() {
 		logger.verbose("Connector | Connection timed out");
 		this._reset();
-		this.emit("close");
+		this.emit("close", lib.ConnectionClosed.Timeout, "Connection timed out");
```

Comment:
I don't think exposing the WebSocket codes in the close event of the connector is a good idea. The user of the connector which is the Link class has no concept of WebSockets, and in some cases uses virtual connectors which doesn't involve WebSockets at all.

The Connector is supposed to be abstraction that handles the WebSockets part for Link so that what Link can use the connector as a way to send messages to the other side.

Reply:
Orginaly the code for killing the host was handled by the host; which meant I needed to pass the close reason on to detect it. However, during development I removed this code and found a cleaner way by emiting errors. Passing the code onwards is a reminent from then which I didn't remove as I suspect it could have a use later. Although you raise a good point for why it should not be included so I'll remove it.

Reply:
Any particular reason why there's still 5 `this.emit` adding the WebSocket codes to the events left in the PR?

Reply:
None, they should have been removed and I missed them.